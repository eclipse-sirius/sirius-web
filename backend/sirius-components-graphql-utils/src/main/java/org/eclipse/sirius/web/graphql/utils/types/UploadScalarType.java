/*******************************************************************************
 * Copyright (c) 2019, 2020 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
package org.eclipse.sirius.web.graphql.utils.types;

import graphql.schema.GraphQLScalarType;

/**
 * Custom {@link GraphQLScalarType} dedicated to Upload.
 *
 * @author SMonnier
 */
public class UploadScalarType {

    public static final String UPLOAD_TYPE = "Upload"; //$NON-NLS-1$

    // @formatter:off
    public static final GraphQLScalarType INSTANCE = GraphQLScalarType.newScalar()
            .name(UPLOAD_TYPE)
            .description("Upload type used for the file upload") //$NON-NLS-1$
            .coercing(new UploadScalarCoercing())
            .build();
    // @formatter:on
}
