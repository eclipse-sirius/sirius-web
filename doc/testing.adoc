# Testing

This document will describe how we should try to write and maintain tests in Sirius Web.

IMPORTANT: TLDR: Use backend integration tests if you don’t have large and complex pieces of frontend code to test that involve the user interface (diagram interactions, diagram layout, form description editor drag and drop, etc).
They can easily be configured to test very complex use cases involving domains, views, custom services, etc.
They are very fast to execute and are very stable.


## What kind of tests do we have?

We have quite a large scope of tests in Sirius Web with:

- Backend unit tests
- Backend integration tests
- Frontend unit tests
- End to end tests
- Manual tests

## What kind of tests should be written?

The kind of tests that should be written depends on the work that has been done and which should be tested and the goal of the tests.

### Backend unit tests

This kind of tests should be used if you have some very complex algorithm in one specific method.
Given the kind of application that we are developing, this kind of tests should be quite rare.
While we had the backend layout algorithm in the past which represented quite a complex set of algorithms with some geometric computation, this is not really the case anymore.
Since most of the complexity we have to deal with is the integration between components, they are not that useful for us today.

### Frontend unit tests

Just like their backend counterparts, such tests should be written for complex pieces of algorithm on the frontend.
Some of our React component can easily have a complex lifecycle using complex DOM interactions and synchronisation steps with the remote server for example.
They could be tested using this approach but most of the time the experience has shown that this may require setting up some complex mocks of both the DOM and the backend.

Most of the complexity in our frontend code comes from the synchronisation of state between user interactions and the lifecycle of the backend.
As a result, most of the code that needs to be tested in the frontend could just as well be tested by end to end tests.

IMPORTANT: The more mocks are necessary to run unit tests, the further the tests are from the reality.
If you need to mock some parts of the database (with mocks of IRepository for backend tests), network interactions like HTTP requests / WebSocket messages (with mocks of useQuery() or useMutation()) or the DOM (jdom), you probably need integration tests instead.


### Backend integration tests

Those tests are to be considered if most of the logic to be tested is located on the backend and if the frontend part of the tested feature is not significant.

They allow you to test complex state of backend business logic.
It is possible to initialize complex state easily with snapshots of the database or custom services to initialize the editing context with domains or view based representation descriptions.
You can write tests to validate the complex steps of the lifecycle of the representations (create a representation, validate some initial state, trigger some tool, validate the impact of said tool).
We can even contribute test-specific services to mock customization provided by specifiers.

If you don’t have complex pieces of frontend code to test, they are the most efficient way to test Sirius Web applications.


### End to end tests with Cypress

End to end tests are a great way to manipulate the application like the end user.
They give you the ability to test features involving complex user interactions and frontend / backend exchanges.
For example, there is nothing like end to end tests to validate some frontend interactions in diagrams.

While they may seem perfect at first, they also have some sizable drawbacks.
End to end tests are very costly to setup.
Since we are limited to what the user interface and the HTTP API can do, it’s very complex to perform tests with complex initial state.
Testing new features of the view DSL for example, require the creation of a studio project and then an instance project and that’s both cumbersome, long and subject to errors.
 
It is also very difficult to test invalid use cases since both the user interface and the HTTP API prevent end users from creating issues.

End to end tests are also the most complex tests to write and maintain.
Since we have to interact with the user interface like end users, we have to write every single step taken by the end user even if they are not relevant for the feature to be tested.

These tests can easily be flaky since we have to carefully detect user interface states.
Countless tests have been invalid in the past because of a failure to detect that a server response had not been received before trying to perform an action.
For example, to create a representation, you need to wait for the response of the server in the new representation dialog otherwise the list of available representation is empty.


### Manual tests

This is THE perfect test environment since you will end up testing things using real human beings.
But since we are testing with humans, there are a couple things to consider.
Manual tests should be used for two main use cases.

Either to ensure the non regression of use cases that are so complex that we did not succeed in writing automated tests.
For example concurrent tests with multiple users.
Or to leverage the ingenuity of real human beings to find bugs instead of preventing regressions.

#### Using human beings to prevent regressions

Most of the time, preventing regressions involve starting from a specific state, executing a specific set of instructions and validating a specific end result.
This can and should be automated instead since:

- They can be complex to maintain since we won't easily know which tests should be updated after a release (other kinds of tests tends to break quickly after an incompatible change)
- They are boring to execute as an end user
- It’s a shame not to perform such specific test in the continuous integration process
- Human being are very good at missing small and boring details

As such using human for non regression is not really efficient. Let’s be more careful with the contribution of non regression tests when pull results are submitted.

#### Using humans to discover new bugs

Instead it is better to use human beings to use the application to find new bugs.
By giving them some freedom in the way each test should be executed, we can easily end up with different testers using different paths.
In order to discover bugs, the difference in the execution path can sometime be as subtle as clicking on menu items in a different order.
Those manual tests shouldn't be too complex or too long to not frustrate testers. 

## Decision

Here are a couple of directions in which we will move with regards to tests:

- If the vast majority of the behavior being tested is performed by the backend and if the frontend is not very complicated, we will default to backend integration tests
- If a sizable part of the behavior being tested is performed in the frontend, we will default to end to end tests or unit tests
- If a manual test can be automated, we will remove them and use an automated test instead
- End to end tests are very time consuming, we will try to restrain ourselves from writing this kind of test if backend integration tests can do the same job
- Flakky end to end tests are eroding our trust in the build by making everyone fine with red builds.
They will be removed and issues will be created to either restore them if the feature tested had to be tested using an end to end test or to convert them to backend integration tests.
