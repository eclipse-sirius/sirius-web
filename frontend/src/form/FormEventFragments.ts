/*******************************************************************************
 * Copyright (c) 2021 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import gql from 'graphql-tag';

export const subscribersUpdatedEventPayloadFragment = gql`
  fragment subscribersUpdatedEventPayloadFragment on SubscribersUpdatedEventPayload {
    id
    subscribers {
      username
    }
  }
`;

export const widgetSubscriptionsUpdatedEventPayloadFragment = gql`
  fragment widgetSubscriptionsUpdatedEventPayloadFragment on WidgetSubscriptionsUpdatedEventPayload {
    id
    widgetSubscriptions {
      widgetId
      subscribers {
        username
      }
    }
  }
`;

export const formRefreshedEventPayloadFragment = gql`
  fragment formRefreshedEventPayloadFragment on FormRefreshedEventPayload {
    id
    form {
      id
      label
      pages {
        id
        label
        groups {
          id
          label
          widgets {
            id
            __typename
            diagnostics {
              id
              kind
              message
            }
            ... on Textfield {
              label
              stringValue: value
            }
            ... on Textarea {
              label
              stringValue: value
            }
            ... on Checkbox {
              label
              booleanValue: value
            }
            ... on Select {
              label
              value
              options {
                id
                label
              }
            }
            ... on Radio {
              label
              options {
                id
                label
                selected
              }
            }
            ... on List {
              label
              items {
                id
                label
                imageURL
              }
            }
          }
        }
      }
    }
  }
`;
