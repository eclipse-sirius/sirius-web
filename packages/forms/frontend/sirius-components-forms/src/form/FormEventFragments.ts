/*******************************************************************************
 * Copyright (c) 2021, 2024 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/

export const commonFields = `
  fragment commonFields on Widget {
    id
    __typename
    diagnostics {
      id
      kind
      message
    }
    hasHelpText
    readOnly
  }
`;

export const widgetFields = `
  ${commonFields}
  
  fragment textfieldFields on Textfield {
    label
    iconURL
    stringValue: value
    supportsCompletion
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment textareaFields on Textarea {
    label
    iconURL
    stringValue: value
    supportsCompletion
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment checkboxFields on Checkbox {
    label
    iconURL
    booleanValue: value
    style {
      color
      labelPlacement
    }
  }

  fragment selectFields on Select {
    label
    iconURL
    value
    options {
      id
      label
      iconURL
    }
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment multiSelectFields on MultiSelect {
    label
    iconURL
    values
    options {
      id
      label
      iconURL
    }
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment radioFields on Radio {
    label
    iconURL
    options {
      id
      label
      selected
    }
    style {
      color
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment listFields on List {
    label
    iconURL
    items {
      id
      label
      kind
      iconURL
      deletable
    }
    style {
      color
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment linkFields on Link {
    label
    iconURL
    url
    style {
      color
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment buttonFields on Button {
    label
    buttonLabel
    imageURL
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment splitButtonFields on SplitButton {
    label
    actions {
      ...commonFields
      ...buttonFields
    }
  }

  fragment toolbarActionFields on ToolbarAction {
    label
    buttonLabel
    imageURL
    style {
      backgroundColor
      foregroundColor
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment chartWidgetFields on ChartWidget {
    label
    chart {
      __typename
      ... on BarChart {
        metadata {
          label
        }
        entries {
          key
          value
        }
        style {
          barsColor
          fontSize
          italic
          bold
          underline
          strikeThrough
        }
        width
        height
      }
      ... on PieChart {
        metadata {
          label
        }
        entries {
          key
          value
        }
        style {
          colors
          strokeColor
          strokeWidth
          fontSize
          italic
          bold
          underline
          strikeThrough
        }
      }
    }
  } 

  fragment labelWidgetFields on LabelWidget {
    label
    stringValue: value
    style {
      color
      fontSize
      italic
      bold
      underline
      strikeThrough
    }
  }

  fragment treeWidgetFields on TreeWidget {
    label
    iconURL
    expandedNodesIds
    nodes {
      id
      parentId
      label
      kind
      iconURL
      endIconsURL
      selectable
      checkable
      value
    }
  }

  fragment imageFields on Image {
    label
    iconURL
    url
    maxWidth
  }

  fragment richTextFields on RichText {
    label
    iconURL
    stringValue: value
  }

  fragment sliderFields on Slider {
    label
    iconURL
    minValue
    maxValue
    currentValue
  }
  
  fragment dateTimeFields on DateTime {
    label
    iconURL
    stringValue
    type
    style {
      backgroundColor
      foregroundColor
      italic
      bold
    }
  }

  fragment widgetFields on Widget {
    ...commonFields
    ... on Textfield {
      ...textfieldFields
    }
    ... on Textarea {
      ...textareaFields
    }
    ... on Checkbox {
      ...checkboxFields
    }
    ... on Select {
      ...selectFields
    }
    ... on MultiSelect {
      ...multiSelectFields
    }
    ... on Radio {
      ...radioFields
    }
    ... on List {
      ...listFields
    }
    ... on Link {
      ...linkFields
    }
    ... on Button {
      ...buttonFields
    }
    ... on SplitButton {
      ...splitButtonFields
    }
    ... on ToolbarAction {
      ...toolbarActionFields
    }
    ... on LabelWidget {
      ...labelWidgetFields
    }
    ... on ChartWidget {
      ...chartWidgetFields
    }
    ... on TreeWidget {
      ...treeWidgetFields
    }
    ... on Image {
      ...imageFields
    }
    ... on RichText {
      ...richTextFields
    }
    ... on Slider {
      ...sliderFields
    }
    ... on DateTime {
      ...dateTimeFields
    }
  }
`;

export const flexboxContainerFields = `
  ${commonFields}
  ${widgetFields}
  fragment flexboxContainerFields on FlexboxContainer {
    ...commonFields
    label
    flexDirection
    flexWrap
    flexGrow
    children {
      ...widgetFields
      ... on FlexboxContainer {
        ...commonFields
        label
        flexDirection
        flexWrap
        flexGrow
        children {
          ...widgetFields
          ... on FlexboxContainer {
            ...commonFields
            label
            flexDirection
            flexWrap
            flexGrow
            children {
              ...widgetFields
            }
            borderStyle {
              color
              lineStyle
              size
              radius
            }
          }
        }
        borderStyle {
          color
          lineStyle
          size
          radius
        }
      }
    }
    borderStyle {
      color
      lineStyle
      size
      radius
    }
  }
`;

export const formRefreshedEventPayloadFragment = `
  ${widgetFields}
  ${flexboxContainerFields}
  fragment formRefreshedEventPayloadFragment on FormRefreshedEventPayload {
    id
    form {
      id
      pages {
        id
        label
        groups {
          id
          label
          displayMode
          toolbarActions {
            ...commonFields
            ...toolbarActionFields
          }
          borderStyle {
            color
            lineStyle
            size
            radius
          }
          widgets {
            ...widgetFields
            ... on FlexboxContainer {
              ...flexboxContainerFields
            }
          }
        }
        toolbarActions {
            ...commonFields
            ...toolbarActionFields
        }
      }
    }
  }
`;
