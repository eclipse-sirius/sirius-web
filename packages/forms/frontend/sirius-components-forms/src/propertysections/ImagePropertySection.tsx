/*******************************************************************************
 * Copyright (c) 2022, 2024 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import { ServerContext, ServerContextValue } from '@eclipse-sirius/sirius-components-core';
import Typography from '@material-ui/core/Typography';
import { Theme, makeStyles } from '@material-ui/core/styles';
import { useContext, useEffect, useState } from 'react';
import { PropertySectionComponent, PropertySectionComponentProps } from '../form/Form.types';
import { GQLImage } from '../form/FormEventFragments.types';
import { ImageStyleProps } from './ImagePropertySection.types';
import { PropertySectionLabel } from './PropertySectionLabel';

const useImageStyles = makeStyles<Theme, ImageStyleProps>(() => ({
  container: {
    display: 'grid',
    gridTemplateColumns: ({ maxWidth }) => {
      if (maxWidth) {
        let max = maxWidth;
        if (maxWidth.match(/[0-9]$/)) {
          max = maxWidth + 'px';
        }
        return `minmax(auto, ${max})`;
      } else {
        return '1fr';
      }
    },
  },
}));

/**
 * Defines the content of a Image property section.
 */
export const ImagePropertySection: PropertySectionComponent<GQLImage> = ({
  editingContextId,
  formId,
  widget,
}: PropertySectionComponentProps<GQLImage>) => {
  const { httpOrigin } = useContext<ServerContextValue>(ServerContext);
  const [validImage, setValidImage] = useState<boolean>(true);

  const onErrorLoadingImage = () => {
    setValidImage(false);
  };

  useEffect(() => {
    setValidImage(true);
  }, [widget.url]);

  let imageURL: string;
  if (widget.url.startsWith('http://') || widget.url.startsWith('https://')) {
    imageURL = widget.url;
  } else {
    imageURL = httpOrigin + widget.url;
  }
  const classes = useImageStyles({
    maxWidth: widget.maxWidth,
  });
  return (
    <div>
      <PropertySectionLabel editingContextId={editingContextId} formId={formId} widget={widget} subscribers={[]} />
      <div className={classes.container}>
        {validImage ? (
          <img id={widget.id} src={imageURL} width="100%" onError={onErrorLoadingImage} />
        ) : (
          <Typography variant="caption">No image</Typography>
        )}
      </div>
    </div>
  );
};
