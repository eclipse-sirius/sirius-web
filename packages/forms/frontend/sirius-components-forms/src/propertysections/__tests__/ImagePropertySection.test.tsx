/*******************************************************************************
 * Copyright (c) 2022, 2024 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
import { MockedProvider } from '@apollo/client/testing';
import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { afterEach, expect, test, vi } from 'vitest';
import { GQLImage } from '../../form/FormEventFragments.types';
import { ImagePropertySection } from '../ImagePropertySection';

crypto.randomUUID = vi.fn(() => '48be95fc-3422-45d3-b1f9-d590e847e9e1');

afterEach(() => cleanup());

const imageWithMaxWidth: GQLImage = {
  label: 'myImage',
  url: 'https://www.eclipse.org/sirius/common_assets/images/logos/logo_sirius.png',
  maxWidth: '42px',
  iconURL: [],
  hasHelpText: false,
  __typename: 'Image',
  diagnostics: [],
  id: 'imageId',
  readOnly: false,
};

const imageWithNoMaxWidth: GQLImage = {
  label: 'myImage',
  url: 'https://www.eclipse.org/sirius/common_assets/images/logos/logo_sirius.png',
  maxWidth: '',
  iconURL: [],
  hasHelpText: false,
  __typename: 'Image',
  diagnostics: [],
  id: 'imageId',
  readOnly: false,
};

test('render image widget with maxWidth', () => {
  const { container } = render(
    <MockedProvider>
      <ImagePropertySection
        editingContextId="editingContextId"
        formId="formId"
        widget={imageWithMaxWidth}
        readOnly={false}
      />
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
  const containerStyle = window.getComputedStyle(document.querySelectorAll('div')[4]);
  expect(containerStyle.display).toEqual('grid');
  expect(containerStyle['grid-template-columns']).toEqual('minmax(auto, 42px)');
});

test('render image widget without maxWidth', () => {
  const { container } = render(
    <MockedProvider>
      <ImagePropertySection
        editingContextId="editingContextId"
        formId="formId"
        widget={imageWithNoMaxWidth}
        readOnly={false}
      />
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
  const containerStyle = window.getComputedStyle(document.querySelectorAll('div')[4]);
  expect(containerStyle.display).toEqual('grid');
  expect(containerStyle['grid-template-columns']).toEqual('1fr');
});

test('render image widget with help hint', () => {
  const { container } = render(
    <MockedProvider>
      <ImagePropertySection
        editingContextId="editingContextId"
        formId="formId"
        widget={{ ...imageWithMaxWidth, hasHelpText: true }}
        readOnly={false}
      />
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
  const containerStyle = window.getComputedStyle(document.querySelectorAll('div')[4]);
  expect(containerStyle.display).toEqual('grid');
  expect(containerStyle['grid-template-columns']).toEqual('minmax(auto, 42px)');
});
