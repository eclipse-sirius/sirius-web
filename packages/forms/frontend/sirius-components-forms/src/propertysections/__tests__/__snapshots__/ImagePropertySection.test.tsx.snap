// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render image widget with help hint 1`] = `
<div>
  <div>
    <div
      class="makeStyles-propertySectionLabel-13"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-14 MuiTypography-subtitle2"
      >
        myImage
      </h6>
      <div
        class="makeStyles-decorators-15"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
          focusable="false"
          style="margin-left: 8px; font-size: 16px;"
          title=""
          viewBox="0 0 24 24"
        >
          <path
            d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="makeStyles-container-11 makeStyles-container-12"
    >
      <img
        id="imageId"
        src="https://www.eclipse.org/sirius/common_assets/images/logos/logo_sirius.png"
        width="100%"
      />
    </div>
  </div>
</div>
`;

exports[`render image widget with maxWidth 1`] = `
<div>
  <div>
    <div
      class="makeStyles-propertySectionLabel-3"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-4 MuiTypography-subtitle2"
      >
        myImage
      </h6>
      <div
        class="makeStyles-decorators-5"
      />
    </div>
    <div
      class="makeStyles-container-1 makeStyles-container-2"
    >
      <img
        id="imageId"
        src="https://www.eclipse.org/sirius/common_assets/images/logos/logo_sirius.png"
        width="100%"
      />
    </div>
  </div>
</div>
`;

exports[`render image widget without maxWidth 1`] = `
<div>
  <div>
    <div
      class="makeStyles-propertySectionLabel-8"
    >
      <h6
        class="MuiTypography-root makeStyles-typography-9 MuiTypography-subtitle2"
      >
        myImage
      </h6>
      <div
        class="makeStyles-decorators-10"
      />
    </div>
    <div
      class="makeStyles-container-6 makeStyles-container-7"
    >
      <img
        id="imageId"
        src="https://www.eclipse.org/sirius/common_assets/images/logos/logo_sirius.png"
        width="100%"
      />
    </div>
  </div>
</div>
`;
